src/index.ts

```diff
@@ -0,0 +1,310 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+import IMachineLearningModel from './IMachineLearningModel';
+
+/** Map from workspaces to BlockSuggestor objects. */
+const suggestorLookup = new WeakMap<Blockly.Workspace, BlockSuggestor>();
+
+/**
+ * Class that tracks all blocks created in a workspace and suggests future
+ * blocks to use.
+ */
+export class BlockSuggestor {
+    /**
+   * Constructs a BlockSuggestor object.
+   * @param {number} numBlocksPerCategory the size of each toolbox category
+   */
+  private description?: string;
+  // Properties with explicit types
+  private model: IMachineLearningModel | null = null;
+    /**
+     * Saves the full JSON data for each block type the first time it's used.
+     * This helps store what initial configuration / sub-blocks each block type
+     * would be expected to have.
+     */
+  private defaultJsonForBlockLookup: Record<
+    string,
+    Blockly.utils.toolbox.BlockInfo
+  > = {};
+      /**
+     * List of reently used block types
+     */
+  private recentlyUsedBlocks: string[] = [];
+      /**
+     * Checks if the workspace is finished loading, to avoid taking action on
+     * all the BLOCK_CREATE events during workspace loading.
+     */
+  private workspaceHasFinishedLoading = false;
```

Comment:
Leave this as `public` or make the `setWorkspaceHasFinishedLoading` function a setter so that https://github.com/jschanker/suggested-blocks-blockly/blob/1a3d44b104acdfb33bea3694f9d137b214e4025d/src/index.ts#L255 can be made what it was originally (https://github.com/jschanker/suggested-blocks-blockly/blob/6f51c711fa3f00d3cebf3a7b55109df7f809a239/src/index.js#L201)